src/MarginBase.sol

```diff
@@ -0,0 +1,801 @@
+
+            unchecked {
+                commandIndex++;
+            }
+        }
+    }
+
+    function _dispatch(Command command, bytes memory inputs) internal {
+        // if-else logic to dispatch commands
+        if (command == Command.PERPS_V2_DEPOSIT) {
+            (address market, int256 amount) = abi.decode(
+                inputs,
+                (address, int256)
+            );
+            _perpsV2Deposit({_market: market, _amount: amount});
+        } else if (command == Command.PERPS_V2_WITHDRAW) {
+            (address market, int256 amount) = abi.decode(
+                inputs,
+                (address, int256)
+            );
+            _perpsV2Withdraw({_market: market, _amount: amount});
+        } else if (command == Command.PERPS_V2_EXIT) {
+            (address market, uint256 priceImpactDelta) = abi.decode(
+                inputs,
+                (address, uint256)
+            );
+            _perpsV2ExitPosition({
+                _market: market,
+                _priceImpactDelta: priceImpactDelta
+            });
+        } else if (command == Command.PERPS_V2_SUBMIT_ATOMIC_ORDER) {
+            (address market, int256 sizeDelta, uint256 priceImpactDelta) = abi
+                .decode(inputs, (address, int256, uint256));
+            _perpsV2SubmitAtomicOrder({
+                _market: market,
+                _sizeDelta: sizeDelta,
+                _priceImpactDelta: priceImpactDelta
+            });
+        } else if (command == Command.PERPS_V2_SUBMIT_DELAYED_ORDER) {
+            (
+                address market,
+                int256 sizeDelta,
+                uint256 priceImpactDelta,
+                uint256 desiredTimeDelta
+            ) = abi.decode(inputs, (address, int256, uint256, uint256));
+            _perpsV2SubmitDelayedOrder({
+                _market: market,
+                _sizeDelta: sizeDelta,
+                _priceImpactDelta: priceImpactDelta,
+                _desiredTimeDelta: desiredTimeDelta
+            });
+        } else if (command == Command.PERPS_V2_SUBMIT_OFFCHAIN_DELAYED_ORDER) {
+            (address market, int256 sizeDelta, uint256 priceImpactDelta) = abi
+                .decode(inputs, (address, int256, uint256));
+            _perpsV2SubmitOffchainDelayedOrder({
+                _market: market,
+                _sizeDelta: sizeDelta,
+                _priceImpactDelta: priceImpactDelta
+            });
+        } else if (command == Command.PERPS_V2_CANCEL_DELAYED_ORDER) {
+            address market = abi.decode(inputs, (address));
+            _perpsV2CancelDelayedOrder({_market: market});
+        } else if (command == Command.PERPS_V2_CANCEL_OFFCHAIN_DELAYED_ORDER) {
+            address market = abi.decode(inputs, (address));
+            _perpsV2CancelOffchainDelayedOrder({_market: market});
+        } else {
+            // placeholder area for further commands
+            revert InvalidCommandType(uint256(command));
+        }
+    }
```

Comment:
Switch statement maybe? Would be a bit more compact

Reply:
switch statement not supported in Solidity ):

(is supported in yul though)